### DISCUSSION TITLE
[Bug]: Altitude problems in Android app and in MQTT

### DISCUSSION THREAD

**Discussion by TheDomanNow** [meshtastic/firmware#4058] [Troubleshooting]
### Category

Other

### Hardware

Heltec V3

### Firmware Version

2.3.10.d19607b

### Description

Hi,

[ANDROID APP > SETTINGS > POSITION CONFIG > Use fixed position]
1. When fixed altitude is set, after device is restarted after some time, the value is being reset to "0", not always but it happens form time to time.
2. MQTT is not passing the altitude value (can see "0" altitude via MQTT).

Heltec V3.1 | Meshtastic Firmware 2.3.10.d19607b Beta | Android app + BLE/IP

BR, Doman.

**Comment by madeofstown** [comment]
What is your GPS Mode setting? I've also struggled to get the GPS position to stick after power cycling on a Pico. What eventually worked for me was to allow the position from my phone to be captured by the node. Then I disable the setting to allow GPS from the phone before reconnecting to the node. After another restart I found that the coordinates hold.

**Comment by TheDomanNow** [comment]
2x ENABLED + Fixed position.

Heltec V3 doesn't have GPS. Devices are not connected to smartphone.

**Comment by madeofstown** [comment]
> Devices are not connected to smartphone.

So I'm assuming that your Android device doesn't have a GPS... 
Have you tried changing the mode setting to "Not Present"? With the setting on "Enabled" I think the device may be looking for a GPS that isn't available could be overwriting your manual entry with empty values.

**Reply by TheDomanNow** [reply]
The issue is; I don't want to pull GPS from phone. I want to use fixed position, this is a base station - GPS is not needed.

For now I've left the setting to NON_PRESENT. On one device this helped, on other it didn't. The one that lost altitude setting was used recently via Web UI (but in general via Android app). I'll observe it more.

**Reply by madeofstown** [reply]
On Android devices pulling GPS data from the phone is a separate setting on the connection panel. If you don't have the box ticked then it won't pull that info from your phone.
![Screenshot_20240611-073255~2](https://github.com/meshtastic/firmware/assets/33820964/5013657e-cdec-4c01-9334-8042a352b36c)

**Comment by TheDomanNow** [comment]
This is a base station, the whole point is that it doesn't have GPS, and I'm using fixed position.

So, for me, setting GPS to NON_PRESENT, and then setting FIXED GPS pos + alt, and enabling FLAGS ALTITUDE + ALTITUDE_MSL worked. I'm not loosing the altitude now, and reporting via MQTT works fine. Case closed.
